testcase/ninja_symlink_outputs.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+#
+# Copyright 2020 Google Inc. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+mk="$@"
+
+cat <<EOF >Makefile
+dangling_symlink:
+	ln -sf nil dangling_symlink
+dangling_symlink: .KATI_SYMLINK_OUTPUTS := dangling_symlink
+
+multiple_symlinks:
+	ln -sf nil symlink_foo && ln -sf nil symlink_bar
+multiple_symlinks: .KATI_SYMLINK_OUTPUTS := symlink_foo symlink_bar
```

Comment:
What are the expected semantics of `.KATI_SYMLINK_OUTPUTS`? Do they declare additional outputs of this particular rule or are they expected to be a subset of the existing outputs?

From this test case, it appears that you intend the semantics to be the former. This is inconsistent with what you did with Ninja:

https://android-review.googlesource.com/c/platform/external/ninja/+/1432871/16/src/build.cc

and then what does it mean if some other rule declares the symlink as a dependency? From the code, it appears that that won't work?

Reply:
No additional semantics. It's just a way to denote the expected outputs of the edge that are symlinks, and for Kati to generate `symlink_outputs = symlink_foo symlink_bar` in the Ninja file.

Reply:
Discussed offline with @lberki: I'll add a check that the KATI_SYMLINK_OUTPUTS are a subset of the expected outputs of the statement, to ensure that this has the same semantics as the Ninja change.

Reply:
PTAL: https://github.com/google/kati/pull/209/files#diff-6104045de99e72eb53aec70f3d0916ff4095194562cff8dd5f4ef03af61f1deaR268-R271

Reply:
Also added a test for the error message.